cli/lib/stats.ts

```diff
@@ -0,0 +1,92 @@
+import { isSameDay, isSameMonth, isSameWeek } from 'date-fns';
+import fetch from 'node-fetch';
+import { readAppdata, saveAppdata } from './appdata';
+
+export enum StatsGroup {
+	STUDIO_CLI_USAGE_UNIQUE = 'studio-cli-usage-unique',
+}
+
+export enum StatsMetric {
+	SUCCESS = 'success',
+	FAILURE = 'failure',
+}
+
+export type AggregateInterval = 'daily' | 'weekly' | 'monthly';
+
+// Bumps a stat if it hasn't been bumped within the current aggregate interval.
+// This allows us to approximate a 1-count-per-user stat without recording which
+// user the event came from (bump stats are anonymous).
+//
+// We don't want to block the thread to record the stat, so this function doesn't
+// await promises before returning.
+export function bumpAggregatedUniqueStat(
```

Comment:
```suggestion
export async function bumpAggregatedUniqueStat(
```
Nit, but this could probably be an async function, right? Would make the function body slightly easier to read if we await'ed instead of then'ed.